src/block/Block.php

```diff
@@ -513,6 +513,15 @@ public function onScheduledUpdate() : void{
 
 	}
 
+	/**
+	 * Called when this block is updated by a state altering projectile in the world.
```

Comment:
This sentence doesn't mean anything to me. It doesn't tell me why an update was triggered, nor what kind of state alterations should take place.

The function is trying to look and sound generic, but it's not.
"interaction" is also too vague.